### DISCUSSION TITLE
Access to XPathEvaluator or DOMParser

### DISCUSSION THREAD

**Discussion by lmcdasi** [mozilla/rhino#1250] [Q&A]
Is it possible to have access to XPathEvaluator or document.evaluate into a java rhino embedded js script ?

I've tried:

`String multiCode="var test=function(text, xPathRegex, xpathResult){" + "\r\n" +
"const parser = new DOMParser();" + "\r\n" +
"const document = parser.parseFromString(text);" + "\r\n" +
"var results=document.evaluate(xPathRegex, text, null, xpathResult, null);" + "\r\n" +
"return results}";
engine.eval(multiCode);`

But the js engine does not have access to DOMParser. I tried XPathEvaluator too.

**Comment by p-bakker** [comment]
Rhino is an EcmaScript/JavaScript engine, which means it doesn't support all the additions that browsers ship with, which is what I think you're looking for

**Reply by rbri** [reply]
HtmlUnit (https://github.com/HtmlUnit/htmlunit) simulates browsers and is based on Rhino - maybe you can find some hints there.

Or maybe you can 'put' you problem in a web page and let HtmlUnit do the work.

**Reply by lmcdasi** [reply]
Would it possible to reference the jquery_xpath.js in the rhino js script and use their evaluate method  ?

I will look at htlmunit too.

Thx.

**Comment by tonygermano** [comment]
Rhino has [e4x](https://www.ecma-international.org/publications-and-standards/standards/ecma-357/) as a built-in xml processor.

You can get xpath support through java.
